src/GafferScene/Instancer.cpp

```diff
@@ -2401,3 +2568,246 @@ void Instancer::PrototypeScope::setPrototype( const EngineData *engine, const Sc
+		{
+
+			// We find the capsules using the engine at shutter open, but the time used to construct the capsules
+			// must be the on-frame time, since the capsules will add their own shutter ( and we also handle
+			// the shutter ourselves for transform matrices )
+			//
+			// For most context variables, we are overwriting them for each prototype anyway, so
+			// we can reuse the context. But timeOffset is relative, so it's important that we reset the
+			// time before we do setPrototypeContextVariables for the next element. ( Should this be more
+			// general instead of assuming that frame is the only variable for which offsetMode may be set? )
+			prototypeScope.setFrame( onFrameTime );
+
+			engines[0]->setPrototypeContextVariables( pointIndex, prototypeScope );
+			auto inserted = prototypeContextsMap.emplace( prototypeScope.context()->hash(), prototypeContextsMap.size() );
+
+			if( inserted.second )
+			{
+				addPrototypeToVectors(
+					prototypesPlug, prototypeRoot, prototypeScope, sampleTimes, capsuleHash, pointIndex, *renderOptions,
+					prototypes, prototypeTransforms, prototypeAttributes
+				);
+			}
+
+			prototypeIndices.push_back( inserted.first->second );
+
+		}
+	}
+	else
+	{
+		addPrototypeToVectors(
+			prototypesPlug, prototypeRoot, scope, sampleTimes, capsuleHash, 0, *renderOptions,
+			prototypes, prototypeTransforms, prototypeAttributes
+		);
+	}
+
+	bool hasAttributes = engines[0]->numInstanceAttributes() > 0;
+
+	// If we don't have per-instance attributes, we can pre-prepare renderer attributes ready to go that
+	// depend only on the prototype
+	std::vector< IECoreScenePreview::Renderer::AttributesInterfacePtr > prototypeRendererAttribs;
+	if( !hasAttributes )
+	{
+		for( IECore::ObjectPtr &a : prototypeAttributes )
+		{
+			CompoundObject* typedA = static_cast< CompoundObject* >( a.get() );
+			prototypeRendererAttribs.push_back( renderer->attributes( typedA ) );
+		}
+	}
+
+
+
+	tbb::parallel_for( tbb::blocked_range<size_t>( 0, instancesForPrototype.size() ), [&]( const tbb::blocked_range<size_t> &r )
+		{
+			vector<M44f> pointTransforms( sampleTimes.size() );
+			std::string name;
+			IECoreScenePreview::Renderer::AttributesInterfacePtr attribsStorage;
+
+			for( size_t idx = r.begin(); idx != r.end(); ++idx )
+			{
+				int pointIndex = instancesForPrototype[idx];
+
+				int prototypeIndex = 0;
+				if( prototypeIndices.size() )
+				{
+					prototypeIndex = prototypeIndices[ idx ];
+				}
+
+				IECoreScenePreview::Renderer::AttributesInterface *attribs;
+				int instanceId = engines[0]->instanceId( pointIndex );
+				if( hasAttributes )
+				{
+					CompoundObjectPtr currentAttributes = new CompoundObject();
+					engines[0]->instanceAttributes( pointIndex, *currentAttributes );
+					for( auto &i : static_cast< CompoundObject* >( prototypeAttributes[prototypeIndex].get() )->members() )
+					{
+						currentAttributes->members()[i.first] = i.second;
+					}
```

Comment:
Looking at `computeBranchAttributes()`, instance attributes should be overriding prototype attributes, rather than the other way round as it is here.